### ISSUE TITLE
[SDK] Support conversion between `Value` and `serde_json::Value` as an individual feature

### ISSUE THREAD

**Issue by sheepduke** [open-feature/rust-sdk#18]
Provider author might fetch JSON response from flag vendors. It would be nice to have an optional feature (call it `serde_json` maybe) that implements `From<serde_json::Value> for crate::Value`.

To make it even more convenient, consider implementing:
```rust
impl<T: serde::de::DeserializeOwned> From<T> for Value
```

Depends on #17

```[tasklist]
### Tasks
```
